ipaserver/install/plugins/update_dna_shared_config.py

```diff
@@ -15,73 +15,95 @@
+            else:
+                return True
         except errors.NotFound:
             logger.error("Could not find DNA plugin entry: %s",
-                         dna_config_base)
-            return False, ()
+                         self.dna_plugin_dn)
+            return False
 
+    def get_shared_cfg(self, plugin_name):
+        dna_config_base = DN(('cn', plugin_name), self.dna_plugin_dn)
         try:
-            entry = conn.get_entry(dna_config_base)
+            entry = self.api.Backend.ldap2.get_entry(dna_config_base)
         except errors.NotFound:
             logger.error("Could not find DNA config entry: %s",
                          dna_config_base)
             return False, ()
+        else:
+            logger.debug('Found DNA config %s', dna_config_base)
 
         sharedcfgdn = entry.single_value.get("dnaSharedCfgDN")
         if sharedcfgdn is not None:
             sharedcfgdn = DN(sharedcfgdn)
+            logger.debug("dnaSharedCfgDN: %s", sharedcfgdn)
+            return sharedcfgdn
         else:
             logger.error(
                 "Could not find DNA shared config DN in entry: %s",
                 dna_config_base)
-            return False, ()
+            return None
+
+    def update_shared_cfg(self, sharedcfgdn, **options):
+        """Update the shared config entry related to that host
+
+        If the shared config entry already exists (like upgrade)
+        the update occurs immediately without sleep.
+
+        If the shared config entry does not exist (fresh install)
+        DS server waits for 30s after its startup to create it.
+        Startup likely occurred few sec before this function is
+        called so this loop will wait for 30s max.
+
+        In case the server is not able to create the entry
+        The loop gives a grace period of 60s before logging
+        the failure to update the shared config entry and return
+        """
+
+        method = options.get('method', "SASL/GSSAPI")
+        protocol = options.get('protocol', "LDAP")
+        dna_bind_method = "dnaRemoteBindMethod"
+        dna_conn_protocol = "dnaRemoteConnProtocol"
 
-        #
-        # Update the shared config entry related to that host
-        #
-        # If the shared config entry already exists (like upgrade)
-        # the update occurs immediately without sleep.
-        #
-        # If the shared config entry does not exist (fresh install)
-        # DS server waits for 30s after its startup to create it.
-        # Startup likely occurred few sec before this function is
-        # called so this loop will wait for 30s max.
-        #
-        # In case the server is not able to create the entry
-        # The loop gives a grace period of 60s before logging
-        # the failure to update the shared config entry and return
-        #
         max_wait = 30
+
+        conn = self.api.Backend.ldap2
         fqdn = self.api.env.host
+
         for _i in range(0, max_wait + 1):
```

Comment:
Again, copied code, but max_wait is assigned then used once, here, and it counts from 0 but adds 1 so this will range 0..30 for 31 iterations, up to 62 seconds.